File: JULY-2025/05-07-2025/p1394.swift
class Solution {
    func findLucky(_ arr: [Int]) -> Int {
        var frequency: [Int: Int] = [:]
        frequency.reserveCapacity(arr.count)

        for num in arr {
            frequency[num, default: 0] += 1
        }

        var maxLuckyNum = -1
        for num in arr {
            if frequency[num] == num && num > maxLuckyNum {
                maxLuckyNum = num
            }
        }

        return maxLuckyNum
    }
}

File: JULY-2025/05-07-2025/p1394.cs
public class Solution {
    public int FindLucky(int[] arr) {
		int n = arr.Length;
		int[] cnt = new int[500];
		foreach(int i in arr){
			cnt[i-1]++;
		}
		int res = -1;
		for(int i=0;i<500;i++){
			if(cnt[i] == i+1){
				res = i+1;
			}
		}
		return res;
    }
}